sunpy/map/mapbase.py

```diff
@@ -635,10 +635,62 @@ def observer_coordinate(self):
             if all([k in self.meta for k in keys]):
                 return SkyCoord(obstime=self.date, **kwargs).heliographic_stonyhurst
 
+        all_meta = []           # List to store all possible metadata from each frame
+        meta_present = {}       # Dictionary storing all keys present for each frame respectively
+        count = {}              # Dictionary storing count of all keys present for each frame
+        key_frame_dict = {}     # Dictionary mapping keys to respective frames
+        frame_key_dict = {}     # Dictionary mapping frames to respective keys
+        missing_dict = {}       # Dictionary storing missing keys for each frame
+        shared_meta = {}        # Dictionary storing meta shared by several frames
+
+        shared_meta['dsun_obs'] = False     # Default value : Not present
+
+        for keys, kwargs in self._supported_observer_coordinates:
+            frame_key_dict[kwargs['frame']] = keys
+            count[kwargs['frame']] = 0
+            for key in keys:
+                key_frame_dict[key] = kwargs['frame']
+                all_meta.append(key)
+
+        for key in self.meta:
+            if key in all_meta:
+                try:
+                    meta_present[key_frame_dict[key]].append(key)
+                except KeyError:
+                    meta_present[key_frame_dict[key]] = [key]
+                count[key_frame_dict[key]] += 1
+                if key in shared_meta.keys():
+                    shared_meta[key] = True
+
+        for frame in count:
+            if count[frame] != 0 and count[frame] != 3:
+                missing_dict[str(frame)] = []
+                for i in frame_key_dict[frame]:
+                    if i not in meta_present[frame] and i not in shared_meta.keys():
+                        missing_dict[str(frame)].append(i)
+
+        missing_str = ""
+
+        if not all(shared_meta[i] for i in shared_meta):
```

Comment:
Is 
```python
not all(shared_meta)
```

not enough?

Reply:
`shared_meta` is to keep track of all the meta that will be shared between frames.
`not all(shared_meta)` would evaluate to true as long as there is no `None` or `False` in the keys.
Here I want to keep track of which meta are present and so, I must evaluate over the key values.

Reply:
Oh. Then 

```python
not all(shared_meta.values())
```

should do